contracts/rust/deployer/src/lib.rs

```diff
@@ -3,16 +3,63 @@ use std::{collections::HashMap, io::Write};
 use alloy::{
     contract::RawCallBuilder,
     hex::{FromHex, ToHexExt},
-    network::TransactionBuilder,
+    network::{Ethereum, EthereumWallet, TransactionBuilder},
     primitives::{Address, Bytes, U256},
-    providers::Provider,
+    providers::{
+        fillers::{FillProvider, JoinFill, WalletFiller},
+        utils::JoinedRecommendedFillers,
+        Provider, ProviderBuilder, RootProvider,
+    },
     rpc::types::TransactionReceipt,
+    signers::local::{coins_bip39::English, MnemonicBuilder, PrivateKeySigner},
+    transports::http::reqwest::Url,
 };
 use anyhow::{anyhow, Result};
 use clap::{builder::OsStr, Parser};
 use derive_more::Display;
 use hotshot_contract_adapter::sol_types::*;
 
+pub mod builder;
+pub mod network_config;
+
+/// Type alias that connects to providers with recommended fillers and wallet
+/// use `<HttpProviderWithWallet as WalletProvider>::wallet()` to access internal wallet
+/// use `<HttpProviderWithWallet as WalletProvider>::default_signer_address(&provider)` to get wallet address
+pub type HttpProviderWithWallet = FillProvider<
+    JoinFill<JoinedRecommendedFillers, WalletFiller<EthereumWallet>>,
+    RootProvider,
+    Ethereum,
+>;
+
+/// a handy thin wrapper around wallet builder and provider builder that directly
+/// returns an instantiated `Provider` with default fillers with wallet, ready to send tx
+#[inline]
```

Comment:
just curious, why `#[inline]`?

Reply:
Compiler is probably smart enough to auto inline, I think this function is too simple to cause JUMP overhead

Reply:
I think we don't need this since we also call these functions very rarely and not a billion times in some tight loop. I don't know what exactly the overhead would be in terms of time but I'm guessing it's on the order of nanoseconds?

Reply:
AFAIK the only time this is useful if you we something to be inlined across crate boundaries and have concrete evidence from profiling or otherwise that it actually makes a difference.

Reply:
removed in [58a21a3](https://github.com/EspressoSystems/espresso-network/pull/3180/commits/58a21a3e236eeb0305869b9326df1012764bd36d)